### PR TITLE
Add docs versioning

### PR DESCRIPTION
**Pull request by kylegach**
>**Note for reviewers**
> 
> This is a much-improved take on #298. This new approach requires _only_ updating releases files and Netlify branch build settings; everything else is automatic (based on the monorepo). 😎 
> 
> I've taken care to organize the commits such that reviewing commit-by-commit should be clear (and doing so is recommended).

### Description

**Add versioning to Storybook docs.**

- Add a workflow to respond to monorepo events dispatched and create version branches
    - Responds to event dispatched by monorepo workflow
    - See: storybookjs/storybook#16580
- Via Netlify's branch deploys, build an instance of the docs portion of the site referencing specific versions from the monorepo.
- Use Netlify's redirects to proxy the versioned URL to the branch build and Netlify's headers to prevent non-latest pages from being indexed by search engines.
- Update URLs throughout to include version (unless latest)
    - Ensure relative links, canonical URLs, and social graph images are not broken
- Add a version selector control to the docs sidebar to allow switching between these versions.
- When not viewing latest docs, add a banner pointing to the latest version.
- Other minor changes
    - Remove version from Header
    - Remove "v" prefix from `latestVersion`
    - Only build addons pages when referencing latest version
    - Add "Publishing new versions" section to README

### Screenshot

Docs page showing the version in the url, version selector in the sidebar, and version callout above the content

### How to Test

1. Check Chromatic tests
1. Run the site locally (`yarn start` & `yarn build`)
    - Ensure the correct version of docs is built and you can navigate fully
        - Keep in mind the redirects won't work
1. Run the site locally with a reference branch specified
    - `BRANCH=release-6-0 yarn build`, or `BRANCH=release-6-0 yarn extract-monorepo-docs && yarn start`
    - Ensure the correct version of docs is built and you can navigate fully
        - Keep in mind the redirects won't work
1. Ensure that `yarn extract-monorepo-docs` and `yarn link-monorepo-docs` still work, with their various arguments
1. Triple-check anything SEO-related (redirect generation code, canonical URLs, headers), please 😉 

#### TODO

- [ ] Ensure algolia search works w/ and w/o versions
    - Fairly sure it works w/o
    - They provided a path for versioning support, but will wait until we get on their latest beta release

After merging this...
- [ ] Add `release-6-0`, `release-6-1`, `release-6-2`, `release-6-4` branches to Netlify build settings

After merging https://github.com/storybookjs/storybook/pull/16580...
- [ ] Push to those branches in the monorepo to kick off the process

### CODE REVIEW

**Comment by kylegach** [comment]
I approved the Chromatic checks myself as the visual changes are identical (literally copy/pasted the code) to those already approved in #298.

File: scripts/move-built-files.sh [hunk 1]
```diff
@@ -0,0 +1,22 @@
+#!/bin/bash -
```
Comment (kylegach) [inline scripts/move-built-files.sh:1]: A version^1 of what we do in https://github.com/chromaui/blog/blob/master/scripts/move-to-build.sh, which is based on: https://answers.netlify.com/t/support-guide-can-i-deploy-multiple-repositories-in-a-single-site/179

[1] This is unfortunately much more complex, due to the fact that we're (ultimately) moving `/public/*` into `/public/docs/<version>`, but `/public/docs` already exists and its contents need preserved.

Comment (kylegach) [reply scripts/move-built-files.sh:1]: @winkerVSbecks — This technique did not work. Instead, it results in serving the index page from the `/docs/<version>` route. You can see that here: https://release-6-4--storybook-frontpage.netlify.app/docs/6.4/

Maybe on versioned branch builds I could update the `netlify.toml` ([docs](https://docs.netlify.com/configure-builds/file-based-configuration/#build-settings)) to serve the site from `/public/docs/<version>`?

Comment (winkerVSbecks) [reply scripts/move-built-files.sh:1]: That's true. Yea it won't work cause we're not serving the entire app from that path. 🤦🏽  Let's talk about this tomorrow AM and come up with another plan.

Comment (kylegach) [reply scripts/move-built-files.sh:1]: Had a thought, so I tried this: https://github.com/storybookjs/frontpage/pull/310/commits/55cfa3ec9ca3ee42763c349a1d5c8e0910b3a006

Interestingly, this (a deploy built from that commit) works: https://61847383eb7e4a00077f7ee6--storybook-frontpage.netlify.app/docs/6.4/docs/6.4/react/get-started/introduction/

Even more interestingly, this (a deploy built from _before_ that commit) **also** works: https://6184680172dea0352ffd920a--storybook-frontpage.netlify.app/docs/6.4/docs/6.4/react/get-started/introduction/

_Mostly capturing these thoughts now, so we can reference in the morning..._

**Comment by kylegach** [comment]
Versioned builds shouldn't have a sitemap.

**Comment by kylegach** [comment]
You can preview the (now working! 🎉 thanks, @winkerVSbecks!) functionality in this deploy preview:

https://6189b19a33924f000713adf1--storybook-frontpage.netlify.app/docs/react/get-started/introduction

Note that only 6.2, 6.3 (latest), and 6.4 (beta) are currently built, so you can only switch between those three for this preview build.

**Pull request merged**